feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -123,6 +123,10 @@ def _execute(self, ctx, *, state_a, state_b, mapping, settings, **inputs):
         dict : dict[str, str]
             Dictionary with paths to work arrays, both forward and reverse, and
             trajectory coordinates for systems A and B.
+
+        Notes
+        -----
+        * Here we assume the mapping is only between ``SmallMoleculeComponent``s.
```

Comment:
Hmm, I see what you mean. I think that's a good option, but I wonder if that hinders the possibility of using this Unit in other protocols that do not necessarily rely on ligands mappings (for example in a protein-mutation protocol). That is, assuming we want to try to reuse the units and potentially not having redundant code to do similar things in each protocol.

Reply:
that's a good point - long term maybe the type check should be that the "Protocol" level rather than the "Unit" level?

short term, I generally take the approach of "tight checks on untested features" and then loosen up later

Anyways, this is not something to deal with here, maybe in a next iteration.

Reply:
I've thought about this a little bit more and yes, you are right. While I'd love to avoid redundant code it is something that will happen when we tackle implementing other protocols and integrating them to this code base. I agree. I'll change the type to be a `LigandAtomMapping` as you suggested, it shouldn't affect things.